### PR TITLE
Editor: Unify the BlockContextualToolbar component between post and site editors

### PR DESCRIPTION
**Pull request by youknowriad**
Related #52632

## What?

We're in a middle of an effort to merge the post and site editors under the @wordpress/editor abstraction. As part of this,  this PR unifies the BlockContextualToolbar component between the post and site editors. 

## Testing Instructions

1- Enable Top Toolbar in both post and site editors
2- Ensure that it looks and behaves the same as trunk.

I did make some CSS shuffling so it would be good to check for potential CSS regressions.

### CODE REVIEW

File: packages/editor/src/components/block-contextual-toolbar/index.js [hunk 1]
```diff
@@ -8,36 +8,39 @@ import classnames from 'classnames';
...
+import { unlock } from '../../lock-unlock';
 
 const { useHasBlockToolbar } = unlock( blockEditorPrivateApis );
 
-function ContextualToolbar( {
-	blockSelectionStart,
-	isCollapsed,
-	toggleCollapse,
-} ) {
-	const blockToolbarRef = useRef();
```
Comment (Mamaduka) [inline packages/editor/src/components/block-contextual-toolbar/index.js]: Good catch. The `ref` had no actual use.

File: packages/edit-post/src/components/header/index.js [hunk 2]
```diff
@@ -107,10 +108,9 @@ function Header( { setEntitiesSavedStatesCallback, initialPost } ) {
 			>
 				<DocumentTools disableBlockTools={ isTextEditor } />
 				{ hasTopToolbar && (
-					<ContextualToolbar
+					<BlockContextualToolbar
 						isCollapsed={ isBlockToolsCollapsed }
-						blockSelectionStart={ blockSelectionStart }
-						toggleCollapse={ handleToggleCollapse }
+						onToggle={ handleToggleCollapse }
```
Comment (Mamaduka) [inline packages/edit-post/src/components/header/index.js]: We can pass `setIsBlockToolsCollapsed` here (like in `edit-site`) and remove `handleToggleCollapse`.

Comment (youknowriad) [reply packages/edit-post/src/components/header/index.js]: Good catch.

**Review by Mamaduka — APPROVED** [review]
Thank you, @youknowriad!

Refactoring tests well for me ✅

File: packages/edit-post/src/components/header/index.js [hunk 3]
```diff
@@ -18,20 +17,26 @@ import { useSelect } from '@wordpress/data';
...
 import PostPublishButtonOrToggle from './post-publish-button-or-toggle';
 import MainDashboardButton from './main-dashboard-button';
-import ContextualToolbar from './contextual-toolbar';
 import { store as editPostStore } from '../../store';
 import { unlock } from '../../lock-unlock';
 
-const { DocumentTools, PostViewLink, PreviewDropdown, PinnedItems, MoreMenu } =
-	unlock( editorPrivateApis );
+const {
+	BlockContextualToolbar,
```
Comment (jeryj) [inline packages/edit-post/src/components/header/index.js]: I know this is what I called it, but I'm not convinced of the name `BlockContextualToolbar`. It's the `<BlockToolbar />`, but allows it to be collapsed. Maybe it should be changed to `<CollapsableBlockToolbar />`?  Without the collapsing functionality, it's just a block toolbar.

**Review by jeryj — APPROVED** [review]
I think the name of the component should get changed, but otherwise this looks great! Glad to see them combined.

File: packages/edit-post/src/components/header/index.js (hunk 3 above)
Comment (youknowriad) [reply packages/edit-post/src/components/header/index.js]: Sure works for me. It's going to be an internal component anyway, once we complete the unification of edit-post and edit-site, a lot of these private APIs won't be exported anymore.

**Pull request merged**